src/app/functions/server/getSubscriberEmails.ts

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Session } from "next-auth";
+import { getSubscriberByEmail } from "../../../db/tables/subscribers.js";
+import { getUserEmails } from "../../../db/tables/emailAddresses.js";
+
+/**
+ * NOTE: new function to replace getUserBreaches
+ *
+ * @param user
+ */
+export async function getSubscriberEmails(
+  user: Session["user"]
+): Promise<string[]> {
+  const emailArray: string[] = [];
+  const subscriber = await getSubscriberByEmail(user.email);
+  (await getUserEmails(subscriber.id)).forEach((er) =>
+    emailArray.push(er.email)
+  );
+
+  // push the primary email
+  emailArray.push(user.email);
```

Comment:
You might just be able to seed this on L17 if you want to save an extra `.push()`:
```js
const emailArray: string[] = [user.email];
```

&hellip; not sure something like this is worth the added complexity:
```js
const subscriber = await getSubscriberByEmail(user.email);
const userEmails = await getUserEmails(subscriber.id);
return userEmails.reduce((acc, er) => acc.concat(er.email), [user.email]);
```

Reply:
Thanks @pdehaan, fixed!